src/Schema/Keywords/Pattern.php

```diff
@@ -38,6 +39,13 @@ public function validate($data, string $pattern) : void
 
         $pattern = sprintf('#%s#', str_replace('#', '\#', $pattern));
 
+        if (
+            function_exists('mb_check_encoding') &&
+            (mb_check_encoding($data, 'UTF-8') || mb_check_encoding($pattern, 'UTF-8'))
```

Comment:
should be both data AND pattern be valid UTF-8 in order to use `u` ?

Reply:
I think if the pattern is invalid UTF-8 that should result in an error instead of siliently not adding the modifier.
If data is not valid UTF-8 it would simply not match the condition, so it should not have any influence on the modifier.

Reply:
> I think if the pattern is invalid UTF-8 that should result in an error

sounds valid. if I understand the spec correctly - pattern should be always valid UTF-8 string. So we can throw schema validation failed here defensively

Reply:
Yes, for the "u" modifier to work, both values must be valid UTF-8, but this is the responsibility of the developer, not the validator, as it seems to me. With this check, we only achieve the expected behavior of the validator when one of the values contains UTF-8 characters.

Reply:
1. Data sent in the request should not influence the pattern that is going to validate that data. Sending invalid UTF-8 will allow someone to influence the validation condition. This should not be allowed.
2. Invalid UTF-8 pattern is an error in the specification and should be handled like that.